### PR TITLE
[dev-launcher] remove dev menu from extra modules

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

The reason for this change is that I encountered a crash related to _accessing_ the `DevMenuManager.appInstance.bridge` - I don't fully understand why it was causing a crash, but it is definitely related to this "module setup" step that occurs when registering via the `extraModulesForBridge`. What would happen is the _RNSafeAreaContextProvider_  module would crash.
You can reproduce this yourself by accessing this bridge value (just accessing it causes the crash, you don't need to do anything else) 

What we really want is a few functions from the dev menu manager, so I've updating the implementation to remove this native module setup stage

# How

Moved the dev menu internal functions to the DevMenuManager, and using the shared instance to call these from the dev launcher native module instead

I've also updated the name for clarity, since there are both menu settings, and dev settings

# Test Plan

- I tested that updating the settings actually worked, and that the settings were persisted over multiple refreshes

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by ajsmth** [comment]
@esamelson I've just pushed an update to fix the dev menu settings like we discussed yesterday by adding the module to the DevLauncher context via `DevLauncherPackageDelegate`  and having the DevLauncher module inspect the settings independent of the status of the DevMenu reactContext - this required some additional changes / refactoring

File: packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/modules/DevLauncherInternalModule.kt [hunk 1]
```diff
@@ -194,6 +192,17 @@ class DevLauncherInternalModule(reactContext: ReactApplicationContext?)
     promise.resolve(null)
   }
 
+  @ReactMethod
+  fun getMenuSettingsAsync(promise: Promise) {
+    promise.resolve(reactApplicationContext.getNativeModule(DevMenuSettings::class.java)?.serialize())
```
Comment (esamelson) [inline packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/modules/DevLauncherInternalModule.kt:197]: this is a bit odd; could we just call into the `DevMenuSettings` module directly from JS? i'm not sure i understand the advantage of the indirection provided by this module.

Comment (ajsmth) [reply packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/modules/DevLauncherInternalModule.kt:197]: total brain fart - didn't realize I could just use DevMenuSettings as a native module - will update iOS to match!

**Review by lukmccall — APPROVED** [review]

File: packages/expo-dev-launcher/bundle/native-modules/__mocks__/DevLauncherInternal.ts [hunk 2]
```diff
@@ -13,3 +13,5 @@ export const addDeepLinkListener = jest.fn().mockImplementation(() => jest.fn())
 export const copyToClipboardAsync = jest.fn().mockResolvedValue(null);
 export const getRecentlyOpenedApps = jest.fn().mockResolvedValue([]);
 export const getCrashReport = jest.fn().mockResolvedValue(null);
+
+export const setMenuSettingsAsync = jest.fn().mockResolvedValue({});
```
Comment (esamelson) [inline packages/expo-dev-launcher/bundle/native-modules/__mocks__/DevLauncherInternal.ts:17]: probably no longer needed?

File: packages/expo-dev-launcher/ios/EXDevLauncherInternal.m [hunk 3]
```diff
@@ -12,6 +12,8 @@
 
 NSString *ON_NEW_DEEP_LINK_EVENT = @"expo.modules.devlauncher.onnewdeeplink";
 
+@import EXDevMenu;
```
Comment (esamelson) [inline packages/expo-dev-launcher/ios/EXDevLauncherInternal.m:15]: also maybe no longer needed?

**Review by esamelson — APPROVED** [review]
🎉

**Pull request merged**